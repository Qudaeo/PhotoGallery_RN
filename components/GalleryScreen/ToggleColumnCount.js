import React from "react";
import {StyleSheet, Text, TouchableOpacity, View} from "react-native";

const ToggleColumnCount = ({appColumnCount, toggleColumnCount, isAppInternetReachable}) => {

    const styles = StyleSheet.create({
        columnToggleButton: {
            position: 'absolute',
            right: 15,
            top: 10,
            zIndex: 100,

            borderWidth: 1,
            borderColor: 'rgba(245,245,245,0.5)',
            backgroundColor: isAppInternetReachable
                ? "rgba(153, 255, 153, 0.7)"
                : "rgba(255, 26, 26, 0.7)",
            alignItems: 'center',
            justifyContent: 'center',
            width: 53,
            height: 53,
            borderRadius: 50,
        }
    })

    return <View style={styles.columnToggleButton}>
        <TouchableOpacity onPress={() => toggleColumnCount()}>
            <Text style={{
                fontSize: 24,
                fontWeight: "bold"
            }}
            >{appColumnCount}</Text>
        </TouchableOpacity>
    </View>
}
export default ToggleColumnCount
