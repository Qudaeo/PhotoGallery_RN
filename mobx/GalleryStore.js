import {makeAutoObservable, runInAction} from 'mobx';
import {galleryAPI} from "../api/api";
import {apiPageSize} from "../common/const";

export default class GalleryStore {

    gallery = []
    currentPage = 0

    response = []

    appColumnCount = 1
    appImagesWidth = null

    detailPhoto = {
        id: null,
        width: null,
        height: null,
        download_url: null
    }

    base64Images = {}

    constructor() {
        makeAutoObservable(this, {}, {autoBind: true})
    }

    async getNextPage() {
        runInAction(() => {
            this.currentPage++
        })

        const response = await galleryAPI.getGallery(this.currentPage, apiPageSize)
        const pageResponseData = response.data

        runInAction(() => {
            this.response.push(...pageResponseData)
            this.gallery.push(...pageResponseData)
        })

        /*
        for (let photo of pageResponseData) {
            const imageDimensions = calcImageDimensions(this.appImagesWidth, photo.width / photo.height)
            const getImageResponse = await galleryAPI.getImage(photo.id, imageDimensions.width, imageDimensions.height)

            runInAction(() => {
                this.base64Images[photo.id] = `data:${getImageResponse.headers['content-type'].toLowerCase()};base64,${encode(getImageResponse.data)}`
            })
        }
         */
    }


    setDetailPhoto(photo) {
        this.detailPhoto = {...photo}
    }

    setAppImagesSize(width) {
        this.appImagesWidth = width
    }

    toggleColumnCount() {
        this.appColumnCount = (this.appColumnCount === 1) ? 2 : 1
    }
}
