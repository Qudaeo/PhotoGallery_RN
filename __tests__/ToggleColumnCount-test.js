/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ToggleColumnCount from "../components/GalleryScreen/ToggleColumnCount";
import GalleryStore from "../mobx/GalleryStore";

it('1. renders correctly', () => {
    const galleryStore = new GalleryStore

    renderer.create(<ToggleColumnCount
        appColumnCount={galleryStore.appColumnCount}
        toggleColumnCount={galleryStore.toggleColumnCount}
        isAppInternetReachable={galleryStore.isAppInternetReachable}
        isFetchingInProgress={galleryStore.isFetchingInProgress}
    />)
})

test('2. toMatchSnapshot 1', () => {
    const tree = renderer.create(<ToggleColumnCount
        toggleColumnCount={() => {
        }}
        isAppInternetReachable={true}
        appColumnCount={1}
        isFetchingInProgress={false}
    />).toJSON()
    expect(tree).toMatchSnapshot()
})

test('3. toMatchSnapshot 2', () => {
    const tree = renderer.create(<ToggleColumnCount
        toggleColumnCount={() => {
        }}
        isAppInternetReachable={false}
        appColumnCount={2}
        isFetchingInProgress={true}
    />).toJSON()
    expect(tree).toMatchSnapshot()
})
