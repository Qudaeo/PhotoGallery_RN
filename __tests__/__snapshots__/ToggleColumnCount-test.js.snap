// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`2. toMatchSnapshot 1 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(153, 255, 153, 0.7)",
      "borderColor": "rgba(180, 180, 180,0.7)",
      "borderRadius": 50,
      "borderWidth": 1,
      "elevation": 10,
      "height": 53,
      "justifyContent": "center",
      "position": "absolute",
      "right": 15,
      "top": 10,
      "width": 53,
      "zIndex": 100,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 25,
          "paddingBottom": 1,
        }
      }
    >
      1
    </Text>
  </View>
</View>
`;

exports[`3. toMatchSnapshot 2 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(255, 26, 26, 0.7)",
      "borderColor": "rgba(180, 180, 180,0.7)",
      "borderRadius": 50,
      "borderWidth": 1,
      "elevation": 10,
      "height": 53,
      "justifyContent": "center",
      "position": "absolute",
      "right": 15,
      "top": 10,
      "width": 53,
      "zIndex": 100,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 25,
          "paddingBottom": 1,
        }
      }
    >
      2
    </Text>
  </View>
</View>
`;
